evap/staff/importers.py

```diff
@@ -361,23 +387,50 @@ def save_users_to_db(self):
             occur because of the data already in the database.
         """
         new_participants = []
+        created_users = []
         with transaction.atomic():
-            users_count = 0
             for (sheet, row), (user_data) in self.associations.items():
                 try:
                     user, created = user_data.store_in_database()
                     new_participants.append(user)
                     if created:
-                        users_count += 1
+                        created_users.append(user)
 
                 except Exception as e:
                     self.errors.append(_("A problem occured while writing the entries to the database."
                                          " The original data location was row %(row)d of sheet '%(sheet)s'."
                                          " The error message has been: '%(error)s'") % dict(row=row+1, sheet=sheet, error=e))
                     raise
-        self.success_messages.append(_("Successfully created %(users)d user(s).") % dict(users=users_count))
+
+        msg = _("Successfully created {} user(s):").format(len(created_users))
+        msg += create_user_list_string_for_message(created_users)
+        self.success_messages.append(mark_safe(msg))
         return new_participants
 
+    def get_user_profile_list(self):
+        new_participants = []
+        for user_data in self.users.values():
+            try:
+                new_participant = UserProfile.objects.get(username=user_data.username)
+            except UserProfile.DoesNotExist:
+                new_participant = user_data.create_user_profile()
+
+            new_participants.append(new_participant)
+
+        return new_participants
+
+    def create_test_success_messages(self):
+        self.success_messages.append(_("The test run showed no errors. No data was imported yet."))
+
+        filtered_users = []
+        for (sheet, row), (user_data) in self.associations.items():
```

Comment:
isn't there a list of user_data lying around that you can use instead of `associations`?